geoips/plugin_registry.py

```diff
@@ -14,11 +14,207 @@
+        if isinstance(name, tuple):
+            try:
+                relpath = self._registered_plugins["yaml_based"]["products"][name[0]][
+                    name[1]]["relpath"]
+                package = self._registered_plugins["yaml_based"]["products"][name[0]][
+                    name[1]]["package"]
+            except KeyError:
+                raise PluginError(
+                    f"Plugin [{name[1]}] doesn't exist under source name [{name[0]}]"
+                )
+            abspath = str(resources.files(package) / relpath)
+            plugin = yaml.safe_load(open(abspath, "r"))
+            plugin_found = False
+            for product in plugin["spec"]["products"]:
+                if product["name"] == name[1] and name[0] in product["source_names"]:
+                    plugin_found = True
+                    plugin = product
+                    break
+            if not plugin_found:
+                raise PluginError(
+                    "There is no plugin that has " + name[1] + " included in it."
+                )
+            plugin["interface"] = "products"
+            plugin["package"] = package
+            plugin["abspath"] = abspath
+            plugin["relpath"] = relpath
+        else:
+            try:
+                relpath = self._registered_plugins["yaml_based"][interface][name][
+                    "relpath"
+                ]
+                package = self._registered_plugins["yaml_based"][interface][name][
+                    "package"
+                ]
+            except KeyError:
+                raise PluginError(
+                    f"Plugin [{name}] doesn't exist under interface [{interface}]"
+                )
+            abspath = str(resources.files(package) / relpath)
+            plugin = yaml.safe_load(open(abspath, "r"))
+            plugin["package"] = package
+            plugin["abspath"] = abspath
+            plugin["relpath"] = relpath
+        if just_info:
+            info_dict = {
+                "interface": interface,
+                "package": package,
+                "abspath": abspath,
+                "relpath": relpath,
+                "name": plugin["name"],
+            }
+            if interface == "products":
+                info_dict["source_names"] = plugin["source_names"]
+                info_dict["product_defaults"] = plugin["product_defaults"]
+            return info_dict
+        else:
+            interface_module = getattr(geoips.interfaces, "products")
+            validated = interface_module.validator.validate(plugin)
+            return interface_module._plugin_yaml_to_obj(name, validated)
+
+    def list_plugins(self, interface):
+        """List the plugins available for an interface ONLY based on the registries.
+
+        This should not load any plugins, just return info from the registries.
+        """
+        plugin_type = self.identify_plugin_type(interface)
+        return self._registered_plugins[plugin_type][interface]
```

Comment:
Maybe break `list_plugins` into two methods:
- list_interface_plugin_names
- list_interface_plugin_info

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/pull/392/files#r1379315304
Does that resolve this conversation?

Reply:
These have been created, alongside a specific get_plugin_info function.